src/mixed-reality/azext_mixed_reality/tests/latest/test_spatial_anchors_account.py

```diff
@@ -0,0 +1,69 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.testsdk import (ScenarioTest, ResourceGroupPreparer)
+from knack.util import CLIError
+
+
+class SpatialAnchorsAccountScenarioTest(ScenarioTest):
+
+    @ResourceGroupPreparer(location='eastus2', parameter_name_for_location='location')
+    def test_spatial_anchors_account_scenario(self, resource_group, location):
+
+        name = self.create_random_name(prefix='cli', length=24)
+
+        self.kwargs.update({
+            'initial': 'az spatial-anchors-account',
+            'name': name,
+            'location': location,
+        })
+
+        try:
+            # Create
+            self.__assert_spatial_anchors_account_not_exist()
+
+            self.__assert_spatial_anchors_account_as_expected('{initial} create -g {rg} -n {name} -l {location}')
+
+            # Read
+            self.__assert_spatial_anchors_account_as_expected('{initial} show -g {rg} -n {name}')
+
+            # Primary Key
+            self.__assert_spatial_anchors_account_keys_work('primary', 'secondary')
+
+            # Secondary Key
+            self.__assert_spatial_anchors_account_keys_work('secondary', 'primary')
+
+            # Delete
+            self.__delete_spatial_anchors_account()
+            self.__assert_spatial_anchors_account_not_exist()
+
+        finally:
+            # Delete is idempotent
+            self.__delete_spatial_anchors_account()
+
+    def __assert_spatial_anchors_account_not_exist(self):
+        for item in self.cmd('{initial} list -g {rg}').get_output_in_json():
+            self.assertNotEqual(self.name, item['name'])
```

Comment:
Is this correct? shouldn't you be comparing self.kwargs['name'] to item['name']
```suggestion
            self.assertNotEqual({name}, item['name'])
```

or 

```suggestion
            self.assertNotEqual(self.kwargs['name'], item['name'])
```

I think this code isn't crashing because no accounts have been created.

Reply:
When you use this method, the command returns an empty dict and so the for loop does not iterate with any items. This is why this code isn't crashing. If you called this method after creating an sa account, you will get an `attributeerror` instead of an `AssertionError` @XiangyuL-Microsoft

Reply:
I run such code after create an Spatial Anchors Account:

`for item in self.cmd('{initial} list -g {rg}').get_output_in_json():
    print('[' + str(item['name']) + ']')`

Never got any error but in the test_results.xml, the system output actually contains the created Spatial Anchors Account name.

Reply:
>                         self.assertNotEqual({name}, item['name'])

directly refer kwarg {name} like this would trigger an azdev style check error.